File: 0290-word-pattern/0290-word-pattern.py
class Solution:
    def wordPattern(self, pattern: str, s: str) -> bool:
        s = s.split(" ")
        dict1 = {}
        dict2 = {}
        if len(pattern) != len(s):
            return False

        for i in range(len(pattern)):
            p1 = pattern[i]
            s1 = s[i]
            if (p1 in dict1 and dict1[p1] != s1) or (s1 in dict2 and dict2[s1]!= p1):
                return False
            dict1[p1] = s1
            dict2[s1] = p1
        return True

File: 0290-word-pattern/0290-word-pattern.java
class Solution {
    public boolean wordPattern(String pattern, String s) {
        HashMap<Character, String> map1 = new HashMap<>();
        HashMap<String, Character> map2 = new HashMap<>();

        String[] st = s.split(" ");
        if (pattern.length() != st.length){
            return false;
        }

        for (int i =0; i< pattern.length(); i++){
            char a = pattern.charAt(i);
            String b = st[i];
            if (map1.containsKey(a) && !map1.get(a).equals(b)){
                return false;
            }
            if (map2.containsKey(b) && !map2.get(b).equals(a)){
                return false;
            }

            map1.put(a,b);
            map2.put(b,a);
        }
        return true;
        
    }
}